src/Microsoft.SqlTools.ServiceLayer/ServiceHost.cs

```diff
@@ -245,62 +245,142 @@ private static ConnectionProviderOptions BuildConnectionProviderOptions()
                     {
-                        Name = "Asynchronous Processing",
+                        Name = "asynchronousProcessing",
                         Description = "Application processing enabled",
                         ValueType = ConnectionOption.ValueTypeBoolean
                     },
                     new ConnectionOption
                     {
-                        Name = "Connect Timeout",
+                        Name = "connectTimeout",
                         Description = "Timeout in seconds when establishing a connection",
                         ValueType = ConnectionOption.ValueTypeNumber,
                         DefaultValue = "15"
                     },
                     new ConnectionOption
                     {
-                        Name = "Current Language",
+                        Name = "currentLanguage",
                         ValueType = ConnectionOption.ValueTypeString
                     },
                     new ConnectionOption
                     {
-                        Name = "Column Encrytion Setting",
+                        Name = "columnEncrytionSetting",
                         ValueType = ConnectionOption.ValueTypeCategory,
                         GroupName = "Security",
                         CategoryValues = new string[] { "Disabled", "Enabled" }
                     },
                     new ConnectionOption
                     {
-                        Name = "Encrypt",
+                        Name = "encrypt",
                         GroupName = "Security",
                         ValueType = ConnectionOption.ValueTypeBoolean
                     },
                     new ConnectionOption
                     {
-                        Name = "Persist Security Info",
+                        Name = "persistSecurityInfo",
                         GroupName = "Security",
                         ValueType = ConnectionOption.ValueTypeBoolean
                     },
                     new ConnectionOption
                     {
-                        Name = "Trust Server Certificate",
+                        Name = "trustServerCertificate",
                         GroupName = "Security",
                         ValueType = ConnectionOption.ValueTypeBoolean
                     },
                     new ConnectionOption
                     {
-                        Name = "Attached DB File Name",
+                        Name = "attachedDBFileName",
                         ValueType = ConnectionOption.ValueTypeString
                     },
                     new ConnectionOption
                     {
-                        Name = "Context Connection",
+                        Name = "contextConnection",
+                        ValueType = ConnectionOption.ValueTypeString
+                    },
+                    new ConnectionOption
+                    {
+                        Name = "port",
```

Comment:
should we also add default values for fields that have known default values?  (e.g. 1433)